### ISSUE TITLE
Dissecting the C codes

### ISSUE THREAD

**Issue by piyanatk** [UKZN-Astronomy/corrcal#17]
@anthony-walters has volunteered to help us dissecting the C codes. (Thanks Tony!) The goal will be to understand what is going on in the codes, add comments, and (perhaps) porting some routines to python.

**Comment by anthony-walters** [comment]
@piyanatk @ronniyjoseph 

Hi guys, I added https://github.com/UKZN-Astronomy/corrcal/commit/92312d627167a87c1a52d073095bb869f6aff93e

In this I add comments to the C code, where I was able. Since I'm not very familiar with the _corrcal_ procedure, these are only partially complete. One can still add descriptions for many of the variable names. This should be straightforward for someone who is familiar with the procedure.

Note: All methods use the void declaration and accept pointers to data, which means they don't return anything, and just operate on the what was passed to them. This is just a C workaround to allow multiple values to be returned (functions can only return a single value in C). The input arguments often include a pointer to the result/output. In some cases the output overwrites the input data. See the "returns" part of the comments for details.
